src/Grids/regular_latitude_longitude_grid.jl

```diff
@@ -41,57 +44,54 @@ function RegularLatitudeLongitudeGrid(FT=Float64; size, latitude, longitude, z,
     TZ = Bounded
     topo = (TX, TY, TZ)
 
-    Nx, Ny, Nz = N = validate_size(TX, TY, TZ, size)
-    Hx, Hy, Hz = H = validate_halo(TX, TY, TZ, halo)
+    Nλ, Nϕ, Nz = N = validate_size(TX, TY, TZ, size)
+    Hλ, Hϕ, Hz = H = validate_halo(TX, TY, TZ, halo)
 
     Lλ = λ₂ - λ₁
     Lϕ = ϕ₂ - ϕ₁
     Lz = z₂ - z₁
 
-    Δϕ = Lλ / Nx
-    Δλ = Lϕ / Ny
+    Δϕ = Lϕ / Nϕ
+    Δλ = Lλ / Nλ
     Δz = Lz / Nz
 
     # Now including halos
-    λF₋ = λ₁ - Hx * Δλ
-    ϕF₋ = ϕ₁ - Hy * Δϕ
+    λF₋ = λ₁ - Hλ * Δλ
+    ϕF₋ = ϕ₁ - Hϕ * Δϕ
     zF₋ = z₁ - Hz * Δz
 
-    λF₊ = λ₂ + Hx * Δλ
-    ϕF₊ = ϕ₂ + Hy * Δϕ
+    λF₊ = λ₂ + Hλ * Δλ
+    ϕF₊ = ϕ₂ + Hϕ * Δϕ
     zF₊ = z₂ + Hz * Δz
 
     λC₋ = λF₋ + Δλ / 2
     ϕC₋ = ϕF₋ + Δϕ / 2
     zC₋ = zF₋ + Δz / 2
 
-    λC₊ = λF₋ + total_extent(TX, Hx, Δλ, Lλ)
-    ϕC₊ = ϕF₋ + total_extent(TY, Hy, Δϕ, Lϕ)
-    zC₊ = zF₋ + total_extent(TZ, Hz, Δz, Lz)
```

Comment:
Note the above lines are incorrect because `total_extent` returns the distance from the first cell face to the last cell face:

https://github.com/CliMA/Oceananigans.jl/blob/95fc44b03796a4f9d8c913c1831cd878ddf42825/src/Grids/grid_utils.jl#L49-L56